File: electron/src/app/models/booking.ts
import { Ledger } from "./ledger";

export class Booking {

    id: number;
    date: Date;
    referenceNumber: string;
    bookingDescription: string;
    ledgerShould: Ledger;
    ledgerHave: Ledger;
    subLedgerShould: Ledger;
    subLedgerHave: Ledger;
    value: number;
    referencePath: string;
    financialYear: string;

    constructor() {}

    public getId(): number {
        return this.id;
    }
}

File: server/src/main/java/models/booking/Booking.java
package models.booking;

import models.basic.BasicModel;
import models.ledger.Ledger;

import java.util.Date;

public interface Booking extends BasicModel {

    public int getId();

    public void setId(int id);

    public Date getDate();

    public void setDate(Date date);

    public String getReferenceNumber();

    public void setReferenceNumber(String referenceNumber);

    public String getBookingDescription();

    public void setBookingDescription(String bookingDescription);

    public Ledger getLedgerShould();

    public void setLedgerShould(Ledger ledgerShould);

    public Ledger getLedgerHave();

    public void setLedgerHave(Ledger ledgerHave);

    public Ledger getSubLedgerShould();

    public void setSubLedgerShould(Ledger subLedgerShould);

    public Ledger getSubLedgerHave();

    public void setSubLedgerHave(Ledger subLedgerHave);

    public double getValue();

    public void setValue(double value);

    public String getReferencePath();

    public void setReferencePath(String referencePath);

    public String getFinancialYear();

    public void setFinancialYear(String financialYear);
}
